context_test.go

```diff
@@ -119,3 +119,86 @@ func TestCurrentCallWithNilResult(t *testing.T) {
 	call := CurrentCall(ctx)
 	assert.Nil(t, call, "Should return nil.")
 }
+
+func getParentContext(t *testing.T) ContextWithHeaders {
+	ctx := context.WithValue(context.Background(), "some key", "some value")
+
+	ctx1, _ := NewContextBuilder(time.Second).
+		SetParentContext(ctx).
+		AddHeader("header key", "header value").
+		Build()
+	return ctx1
+}
+
+func TestContextBuilderParentContextNoHeaders(t *testing.T) {
+	ctx := getParentContext(t)
+	assert.EqualValues(t, "header value", ctx.Headers()["header key"])
```

Comment:
here and below, compare the entire map, and use `assert.Equal`, shouldn't need `EqualValues` (that's only for type mismatches).

```go
assert.Equal(t, map[string]string{"header key": "header value"}, ctx.Headers())
```